### DISCUSSION TITLE
Using RoboTHOR scenes with iTHOR API.

### DISCUSSION THREAD

**Discussion by sagadre** [allenai/ai2thor#966] [Q&A] [answered]
I was wondering if there is a way to use iTHOR API functions with RoboTHOR scenes. This seems to be possible based on the demo here: https://ai2thor.allenai.org/demo

However, when I call iTHOR functions they seem to fail. I even tried initializing the RoboTHOR scene by tacking on `_physics` on the end, but that did not seem to work.

I am specifically interested in `InitialRandomSpawn` in RoboTHOR.

Thanks!

**Comment by winthos** [comment, marked as answer]
That specific action, `InitialRandomSpawn` requires some updates to the physics of the RoboTHOR scenes to work. Currently, all RoboTHOR scenes are completely static, so objects do not have the full physics capabilities of the iTHOR scenes, and because of this repositioning objects using actions like `Push`, `Pull` and even `InitialRandomSpawn` are incompatible.

We are working on updating the RoboTHOR scenes to have full physics capabilities, which will allow these actions to be used. I can't give you an exact timeline of when this update will be available, but it is one of our active projects and I will update this discussion once the update is live.
